### ISSUE TITLE
Support for MQTT-over-Secure-WebSockets

### ISSUE THREAD

**Issue by ChristianWeyer** [wolfSSL/wolfMQTT#37]
Hey all!

This looks like a fantastic lib :-)
Are you planning to add support for MQTT-over-Secure-WebSockets?

Thanks.

**Comment by dgarske** [comment]
Hi Christian,

I've added `MQTT over Websockets` to our feature request list. You are the first to ask about supporting it.

Using our wolfSSL library for TLS and our lightweight HTTP client (code here https://github.com/wolfSSL/wolfssl/blob/master/src/wolfio.c#L585 enabled with HAVE_HTTP_CLIENT), this seems fairly easy to implement.

Does your project have a requirement to use MQTT over Websockets?

Thanks,
David Garske, wolfSSL

**Comment by ChristianWeyer** [comment]
Yes. We need to make sure that we can pass any home router setup with an outgoing connection talking to IlT Hub. Thus using WSS in 443 would be quite safe here.

**Comment by deandob** [comment]
I have a similar requirement to Christian (outgoing connection from home router) where WSS is the preferred protocol for MQTT to use.
Also - thanks for the work on this lib for embedded use, very useful.

**Comment by lygstate** [comment]
gentle ping for this
